connectors/microsoft-teams/src/main/java/io/camunda/connector/operation/channel/DocumentHandler.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. Licensed under a proprietary license.
+ * See the License.txt file for more information. You may not use this file
+ * except in compliance with the proprietary license.
+ */
+package io.camunda.connector.operation.channel;
+
+import com.microsoft.graph.core.models.UploadResult;
+import com.microsoft.graph.core.tasks.LargeFileUploadTask;
+import com.microsoft.graph.drives.item.items.item.createuploadsession.CreateUploadSessionPostRequestBody;
+import com.microsoft.graph.models.ChatMessage;
+import com.microsoft.graph.models.ChatMessageAttachment;
+import com.microsoft.graph.models.DriveItem;
+import com.microsoft.graph.models.DriveItemUploadableProperties;
+import com.microsoft.graph.models.ItemReference;
+import com.microsoft.graph.models.UploadSession;
+import com.microsoft.graph.serviceclient.GraphServiceClient;
+import io.camunda.connector.api.error.ConnectorException;
+import io.camunda.connector.api.error.ConnectorInputException;
+import io.camunda.connector.model.request.data.SendMessageToChannel;
+import io.camunda.document.Document;
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.CancellationException;
+
+public class DocumentHandler {
+
+  SendMessageToChannel model;
+
+  public DocumentHandler(SendMessageToChannel model) {
+    this.model = model;
+  }
+
+  public void handleDocuments(GraphServiceClient graphClient, ChatMessage chatMessage) {
+    List<String> fileUrls =
+        model.documents().stream()
+            .filter(Objects::nonNull)
+            .map(
+                document -> {
+                  try {
+                    return uploadDocument(
+                        graphClient, model.groupId(), document, document.metadata().getFileName());
+                  } catch (IOException
+                      | InvocationTargetException
+                      | IllegalAccessException
+                      | NoSuchMethodException e) {
```

Comment:
nit: wdyt if we handle these exceptions inside the `uploadDocument` method, so that we have all exceptions caught on the same level? The stream will also look nicer here

Reply:
Good point thanks. I moved it.